lib/widgets/files_view.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:files/widgets/grid.dart';
+import 'package:files/widgets/table.dart';
+import 'package:files/widgets/workspace.dart';
+import 'package:flutter/cupertino.dart';
+
+class FilesView extends StatelessWidget {
+  final List<FileCell> gridDelegates;
+  final List<FilesRow> tableDelegates;
```

Comment:
that's not what i meant with delegates btw, if you don't feel like doing it you can just implement the FileTable with the FilesRow widget and i can do the rest of the work no issue